scm-diff-editor/src/lib.rs

```diff
@@ -440,24 +440,28 @@ fn print_dry_run(write_root: &Path, state: RecordState) {
         let file_path = write_root.join(file.path.clone());
         let (selected_contents, _unselected_contents) = file.get_selected_contents();
         match selected_contents {
-            SelectedContents::Absent => {
+            FileState::Absent => {
                 println!("Would delete file: {}", file_path.display())
             }
-            SelectedContents::Unchanged => {
-                println!("Would leave file unchanged: {}", file_path.display())
-            }
-            SelectedContents::Binary {
-                old_description,
-                new_description,
-            } => {
-                println!("Would update binary file: {}", file_path.display());
-                println!("  Old: {:?}", old_description);
-                println!("  New: {:?}", new_description);
-            }
-            SelectedContents::Present { contents } => {
-                println!("Would update text file: {}", file_path.display());
-                for line in contents.lines() {
-                    println!("  {line}");
+            FileState::Present { contents, mode: _ } => {
```

Comment:
In theory I suppose it would be possible to print something about the mode change now, but I've left it out